pennylane/transforms/sign_expand/sign_expand.py

```diff
@@ -168,26 +168,27 @@ def construct_sgn_circuit(  # pylint: disable=too-many-arguments
     coeffs = hamiltonian.data
     tapes = []
     for mu, time in zip(mus, times):
-        with tape.__class__() as new_tape:
-            # Put state prep and ansatz on tape in the 'old' register
-            for op in tape.operations:
-                op.queue()
-
-            # Put QSP and Hadamard test on the two ancillas Target and Control
-            qml.Hadamard(controls[0])
-            for i, phi in enumerate(phis):
-                qml.CRX(phi, wires=controls)
-                if i == len(phis) - 1:
-                    qml.CRY(np.pi, wires=controls)
-                else:
-                    evolve_under(hamiltonian.ops, coeffs, 2 * time, controls)
-                    qml.CRZ(-2 * mu * time, wires=controls)
-            qml.Hadamard(controls[0])
-
-            if tape.measurements[0].return_type == qml.measurements.Expectation:
-                qml.expval(-1 * qml.PauliZ(controls[0]))
+        added_operations = []
+        # Put QSP and Hadamard test on the two ancillas Target and Control
+        added_operations.append(qml.Hadamard(controls[0]))
+        for i, phi in enumerate(phis):
+            added_operations.append(qml.CRX(phi, wires=controls))
+            if i == len(phis) - 1:
+                added_operations.append(qml.CRY(np.pi, wires=controls))
             else:
-                qml.var(qml.PauliZ(controls[0]))
+                for ops in evolve_under(hamiltonian.ops, coeffs, 2 * time, controls):
+                    added_operations.extend(ops)
+                added_operations.append(qml.CRZ(-2 * mu * time, wires=controls))
+        added_operations.append(qml.Hadamard(controls[0]))
+
+        operations = tape.operations + added_operations
```

Comment:
I don't think it is a problem, because queuing should not be necessary, I think queuing should only happen when interfacing with the user QNode direcly